src/main/java/frc/team2412/robot/Subsystems/IndexerSubsystem.java

```diff
@@ -0,0 +1,257 @@
+package frc.team2412.robot.Subsystems;
+
+import com.revrobotics.CANSparkMax;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.SpeedControllerGroup;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.team2412.robot.RobotMap;
+import frc.team2412.robot.Commands.IndexerCommands.IntakeBalls;
+import frc.team2412.robot.Commands.IndexerCommands.ProcessBalls;
+import frc.team2412.robot.Commands.IndexerCommands.ShootBalls;
+import frc.team2412.robot.Commands.IndexerCommands.SwitchBalls;
+
+public class IndexerSubsystem extends SubsystemBase {
+	public CANSparkMax indexBackMotor, indexFrontMotor, indexMidMotor;
+	public SpeedControllerGroup indexMotors, indexSideMotors;
+
+	public DigitalInput back, backMid, mid, frontMid, front, intakeFront, intakeBack;
+
+	// NUMBER OF BALLS IN SYSTEM
+	public int numBalls = 0;
+
+	// SIDE OF INDEXER WITH LESS BALLS
+	public IntakeDirection ballUnbalancedSide;
+
+	// to run function once until reset
+	private int run = 0;
+
+	public IndexerSubsystem() {
+		// Motors
+		indexBackMotor = RobotMap.indexFrontMotor;
+		indexFrontMotor = RobotMap.indexBackMotor;
+		indexMidMotor = RobotMap.indexMidMotor;
+		indexMotors = new SpeedControllerGroup(indexFrontMotor, indexBackMotor, indexMidMotor);
+		indexSideMotors = new SpeedControllerGroup(indexFrontMotor, indexBackMotor);
+
+		// Sensors
+		back = RobotMap.back;
+		backMid = RobotMap.backMid;
+		mid = RobotMap.mid;
+		frontMid = RobotMap.frontMid;
+		front = RobotMap.front;
+
+		intakeFront = RobotMap.intakeFront;
+		intakeBack = RobotMap.intakeBack;
+
+		setDefaultCommand(new ProcessBalls(this, new ShootBalls(this), new IntakeBalls(this), new SwitchBalls(this)));
+	}
+
+	// shifters
+
+	// Motors
+	public void stopAll() {
+		indexMotors.set(0.0);
+	}
+
+	public void stopSides() {
+		indexSideMotors.set(0.0);
+	}
+
+	// Sensors
+	public boolean allSensorsOn() {
+		if (back.get() && backMid.get() && mid.get() && frontMid.get() && front.get()) {
+			return true;
+		}
+
+		return false;
+		// check if all sensors are on
+	}
+
+	public boolean allSensorsOff() {
+		if (!back.get() && !backMid.get() && !mid.get() && !frontMid.get() && !front.get()) {
+			return true;
+		}
+
+		return false;
+		// check if all sensors are on
+	}
+
+	public void shoot() {
+		// allShiftDown();
+		indexMidMotor.set(1);
+		if (ballUnbalancedSide == IntakeDirection.FRONT) {
+			indexBackMotor.set(-1);
+		} else {
+			indexFrontMotor.set(1);
+		}
+		if (frontMid.get() && front.get()) {
+			indexFrontMotor.set(0);
+			ballUnbalancedSide = IntakeDirection.FRONT;
+		} else if (backMid.get() && back.get()) {
+			indexBackMotor.set(0);
+			ballUnbalancedSide = IntakeDirection.BACK;
+		} else if (allSensorsOff()) {
+			stopAll();
+			ballUnbalancedSide = IntakeDirection.NONE;

```

Comment:
Should set `numBalls` to zero, as presumably we've now shot them all.  Similarly, the above code should update `numBalls` to match reality as they exit the indexer subsystem and into the shooter.